uninstall-thin-edge_io.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/sh
+
+set -e
+
+usage() {
+    cat <<EOF
+USAGE:
+    delete-thin-edge_io [COMMAND]
+    
+COMMANDS:
+    remove     Uninstall thin-edge.io with keeping configuration files
+    purge      Uninstall thin-edge.io and also remove configuration files  
+
+EOF
+}
+
+stop_a_service_if_running() {
+    status=$(sudo systemctl is-active "$1") && returncode=$? || returncode=$?
+    if [ "$status" = "active" ]; then
+        sudo systemctl stop "$1"
+    fi
+}
+
+stop_extension_services() {
+    stop_a_service_if_running "tedge-watchdog.service"
+    stop_a_service_if_running "tedge-mapper-collectd.service"
+    stop_a_service_if_running "c8y-log-plugin.service"
+    stop_a_service_if_running "c8y-configuration-plugin.service"
+}
+
+remove_or_purge_package_if_exists() {
+    status=$(dpkg -s "$2" | grep -w installed) && returncode=$? || returncode=$?
+    if [ "$status" = "Status: install ok installed" ]; then
+        sudo apt --assume-yes "$1" "$2"
+    fi
+}
+
+# Here don't need to remove the tedge_mapper, tedge_agent, and tedge_watchdog packages explicitly,
+# as they will be removed by removing the tedge package.
+remove_packages() {
```

Comment:
If this function could be parametrized, to accept "purge" or "remove" as arguments, then we could avoid the code duplication between this function and `purge_packages` function.

Reply:
The common factors are already abstracted in `remove_or_purge_package_if_exists`. So, I feel the existing ones are ok.

Reply:
Currently there are only a handful of packages/services, which isn't a big deal. My concern was with future additions. Whenever we add a new plugin in future, we'll end up adding it to these 2 functions. So, the refactoring suggestion was just to avoid that in future rather than the little code duplication that exists today.